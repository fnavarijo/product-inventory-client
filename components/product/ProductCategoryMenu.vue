<template>
  <div>
    <div class="py-3 border-b border-gray-500">
      Categorias
    </div>
    <ul class="mt-2">
      <li
        v-for="(category, index) in categories"
        :key="index"
        class="pl-2 py-3 cursor-pointer text-sm hover:bg-dark-gray-700 hover:bg-opacity-75 hover:text-white transition duration-100 ease-linear"
        @click="select(category)"
      >
        {{ category }}
      </li>
    </ul>
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';

export default Vue.extend({
  name: 'ProductCategoryMenu',
  props: {
    categories: {
      type: Array as PropType<Array<String>>,
      default: [],
    },
  },
  methods: {
    select (option: string): void {
      this.$emit('select', option);
    },
  },
});
</script>
