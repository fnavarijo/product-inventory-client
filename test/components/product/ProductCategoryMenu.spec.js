import { mount } from '@vue/test-utils';

import ProductCategoryMenu from '@/components/product/ProductCategoryMenu.vue';

describe('ProductCategoryMenu component', () => {
  it('should emit selected option on select', async () => {
    const wrapper = mount(ProductCategoryMenu, {
      propsData: {
        categories: [
          'Accesories',
          'Clothing',
          'Shoes',
        ],
      },
    });

    const categoryElement = wrapper.find('li:nth-child(2)');
    await categoryElement.trigger('click');

    expect(wrapper.emitted('select')[0]).toEqual(['Clothing']);
  });

  it('should match snapshot', () => {
    const wrapper = mount(ProductCategoryMenu, {
      propsData: {
        categories: [
          'Accesories',
          'Clothing',
          'Shoes',
        ],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
