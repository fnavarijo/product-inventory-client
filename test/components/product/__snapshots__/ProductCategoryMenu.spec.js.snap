// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCategoryMenu component should match snapshot 1`] = `
<div>
  <div class="py-3 border-b border-gray-500">
    Categorias
  </div>
  <ul class="mt-2">
    <li class="pl-2 py-3 cursor-pointer text-sm hover:bg-dark-gray-700 hover:bg-opacity-75 hover:text-white transition duration-100 ease-linear">
      Accesories
    </li>
    <li class="pl-2 py-3 cursor-pointer text-sm hover:bg-dark-gray-700 hover:bg-opacity-75 hover:text-white transition duration-100 ease-linear">
      Clothing
    </li>
    <li class="pl-2 py-3 cursor-pointer text-sm hover:bg-dark-gray-700 hover:bg-opacity-75 hover:text-white transition duration-100 ease-linear">
      Shoes
    </li>
  </ul>
</div>
`;
